lib/widgets/lightbox.dart

```diff
@@ -89,6 +93,157 @@ class _CopyLinkButton extends StatelessWidget {
+//         final scaffoldMessenger = ScaffoldMessenger.of(context);
+//         String message = zulipLocalizations.lightboxDownloadImageFailed;
+
+//         try {
+//           // Fetch the image with a timeout
+//           final response = await http.get(url).timeout(
+//             const Duration(seconds: 30),
+//             onTimeout: () {
+//               throw TimeoutException("timed out");
+//             },
+//           );
+//           if (response.statusCode == 200) {
+//             // Get the external storage directory
+//             final directory = await getExternalStorageDirectory();
+//             if (directory == null) {
+//               message = zulipLocalizations.lightboxDownloadImageError;
+//             } else {
+//               final downloadPath = '${directory.path.split("Android")[0]}Download';
+
+//               // Create the Downloads folder if it doesn't exist
+//               final downloadFolder = Directory(downloadPath);
+//               if (!await downloadFolder.exists()) {
+//                 await downloadFolder.create(recursive: true);
+//               }
+
+//               final fileName = url.pathSegments.last;
+//               final filePath = '$downloadPath/$fileName';
+
+//               final file = File(filePath);
+//               await file.writeAsBytes(response.bodyBytes);
+
+//               // Trigger Media Scanner so it reflects in the gallery.
+//               await platform.invokeMethod('scanFile', {'path': filePath});
+
+//               message = zulipLocalizations.lightboxDownloadImageSuccess;
+//             }
+//           } else {
+//             message = zulipLocalizations.lightboxDownloadImageFailed;
+//           }
+//         } catch (e) {
+//           if (e is TimeoutException || e is SocketException) {
+//             message = zulipLocalizations.lightboxDownloadImageError;
+//           } else {
+//             message = zulipLocalizations.lightboxDownloadImageError;
+//           }
+//         }
+
+//         // Show a SnackBar notification
+
+//         scaffoldMessenger.showSnackBar(
+//             SnackBar(behavior: SnackBarBehavior.floating, content: Text(message)),
+//         );
+//       }
+//     );
+//   }
+// }
+
+class _DownloadImageButton extends StatelessWidget {
+  const _DownloadImageButton({required this.url});
+
+  final Uri url;
+
+  static const platform = MethodChannel('gallery_saver');
+
+  @override
+  Widget build(BuildContext context) {
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    return IconButton(
+      tooltip: zulipLocalizations.lightboxDownloadImageTooltip,
+      icon: const Icon(Icons.download),
+      onPressed: () async {
+        final scaffoldMessenger = ScaffoldMessenger.of(context);
+        String message = zulipLocalizations.lightboxDownloadImageFailed;
+
+        try {
+          // Fetch the image with a timeout
+          final response = await http.get(url).timeout(
```

Comment:
Will this work if the image is protected by authentication to the Zulip org? What does zulip-mobile do?